app/workers/stale_issue_marker.rb

```diff
@@ -5,68 +5,134 @@ class StaleIssueMarker
+  end
+
+  def unpinned_query_filter
+    " #{PINNED_LABELS.map { |label| %(-label:"#{label}") }.join(" ")}"
+  end
+
+  def validate_repo_has_label(repo, label)
+    unless GithubService.valid_label?(repo, label)
+      raise "The label '#{label}' does not exist on #{repo}"
     end
   end
 
   def mark_as_stale(issue)
-    return if issue.labels.include?(STALE_LABEL_NAME)
+    return if issue.stale?
 
-    validate_repo_has_stale_label(issue.fq_repo_name)
+    validate_repo_has_label(issue.fq_repo_name, stale_label)
+    issue.add_labels([stale_label])
+  end
+
+  def mark_as_requires_triage(issue)
+    return if issue.requires_triage?
+
+    triage_label = GithubService::Issue::REQUIRES_TRIAGE
+
+    validate_repo_has_label(issue.fq_repo_name, triage_label)
+    issue.add_labels([triage_label])
+    issue.remove_label(GithubService::Issue::VERIFIED_LABEL)
+  end
+
+  def comment_as_stale(issue)
+    mark_as_stale(issue)
+
+    message = "This #{issue.type} has been automatically marked as stale " \
+              "because it has not been updated for at least 3 months.\n\n"
+    if issue.pull_request?
+      message << "If these changes are still valid, please remove the "    \
+                 "`stale` label, make any changes requested by reviewers " \
+                 "(if any), and ensure that this issue is being looked "   \
+                 "at by the assigned/reviewer(s)\n\n"
+    else
+      message << "If you can still reproduce this issue on the current " \
+                 "release or on `master`, please reply with all of the " \
+                 "information you have about it in order to keep the "   \
+                 "issue open.\n\n"
+    end
+    message << COMMENT_FOOTER
 
     logger.info("[#{Time.now.utc}] - Marking issue #{issue.fq_repo_name}##{issue.number} as stale")
-    issue.add_labels([STALE_LABEL_NAME])
-    issue.add_comment(STALE_ISSUE_MESSAGE)
+    issue.add_comment(message)
+  end
+
+  def comment_and_close(issue)
+    mark_as_stale(issue)
+
+    message  = "This #{issue.type} has been automatically closed because it " \
+               "has not been updated for at least 3 months.\n\n"
+    message << "Feel free to reopen this #{issue.type} if "
+    message << "these changes are still valid.\n\n" if issue.pull_request?
+    message << "this issue is still valid.\n\n"     unless issue.pull_request?
+    message << COMMENT_FOOTER
+
+    logger.info("[#{Time.now.utc}] - Closing stale PR #{issue.fq_repo_name}##{issue.number}")
+    GithubService.close_pull_request(issue.fq_repo_name, issue.number)
+    issue.add_comment(message)
   end
 
-  def close_pr(pull_request)
-    validate_repo_has_stale_label(pull_request.fq_repo_name)
-    logger.info("[#{Time.now.utc}] - Closing stale PR #{pull_request.fq_repo_name}##{pull_request.number}")
-    GithubService.close_pull_request(pull_request.fq_repo_name, pull_request.number)
-    pull_request.add_comment(CLOSABLE_PR_MESSAGE)
+  def comment_copy_footer
```

Comment:
empty method?

Reply:
Err... might have been me working out a few ideas, and then fudging it when I committed it (or a failed rebase).

Will look into it.

Reply:
Fixed